relayer/chain/substrate/writer.go

```diff
@@ -116,7 +119,7 @@ func (wr *Writer) writeLoop(ctx context.Context) error {
 }
 
 // Write submits a transaction to the chain
-func (wr *Writer) write(_ context.Context, c types.Call) error {
+func (wr *Writer) write(ctx context.Context, c types.Call) error {
 	ext := types.NewExtrinsic(c)
 
 	era := types.ExtrinsicEra{IsImmortalEra: true}
```

Comment:
Would you design still be compatible if we change the extrinsic [mortality](https://wiki.polkadot.network/docs/en/build-protocol-info#transaction-mortality) send mortal extrinsics instead of immortal ones?

Currently, we're sending immortal extrinsics which from my understanding will never be dropped from the txpool unless it runs out of resources or suffers some other abnormal error.

We do need to change over to mortal transactions at some point, and ideally sometime before we deliver the project. I heard immortal extrinsics can have security complications.

Reply:
Immortal extrinsics are an issue indeed: Since nonces are tracked per account, and accounts can be reaped (nonce will be reset), any immortal transaction can be replayed.

Reply:
Hence, anyone can mess up our bridge by replaying immortal extrinsics after a reap happens (even if accidentally).

Reply:
ah, good to know the exact reason why its harmful. I've updated our ticket https://github.com/Snowfork/polkadot-ethereum/issues/76 for after looking into how we can fix the problem.

Reply:
> Would your design still be compatible if we change the extrinsic mortality send mortal extrinsics instead of immortal ones?

Yeah, I believe so. Regardless, now's a good time to make the switch and test. I'll update this PR